packages/components/src/internal/components/forms/input/useDataChangeCommentsRequired.ts

```diff
@@ -0,0 +1,24 @@
+import { useEffect, useState } from 'react';
+import { useServerContext } from '../../base/ServerContext';
+import { isProductProjectsEnabled, resolveModuleContext } from '../../../app/utils';
+import { useAppContext } from '../../../AppContext';
+
+export const useDataChangeCommentsRequired = (): { requiresUserComment: boolean } => {
```

Comment:
I appreciate the idea of using a hook to encapsulate this state, however, it looks like this is exclusively utilized with a `CommentTextArea`. Should this logic be internalized to that component instead? That would simplify error handling, loading state, etc.

Reply:
I started out encapsulating this into `CommentTextArea` but realized I need to know this setting outside of that component so I can know when to enable or disable the various submit buttons when the comment is required.